src/plonk.rs

```diff
@@ -74,6 +77,48 @@ impl<C: CurveAffine> VerifyingKey<C> {
             cs,
         })
     }
+
+    /// Hashes a verification key into a transcript.
+    pub fn hash_into<T: Transcript<C>>(&self, transcript: &mut T) -> io::Result<()> {
+        let mut hasher = Blake2bParams::new()
+            .hash_length(64)
+            .personal(C::BLAKE2B_PERSONALIZATION)
+            .to_state();
+
+        let s = format!("{}", self);
+
+        hasher.update(&(s.len() as u64).to_le_bytes());
+        hasher.update(s.as_bytes());
+
+        // Hash in final Blake2bState
+        transcript.common_scalar(C::Scalar::from_bytes_wide(
+            hasher.finalize().as_bytes().try_into().unwrap(),
+        ))?;
+
+        Ok(())
+    }
+}
+
+impl<C: CurveAffine> std::fmt::Display for VerifyingKey<C> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> Result<(), std::fmt::Error> {
+        write!(
+            f,
+            "VerificationKey {{\
+            base_modulus = {base_modulus}, \
+            scalar_modulus = {scalar_modulus}, \
+            domain = {domain}, \
+            cs = {cs}, \
+            fixed_commitments = {fixed_commitments:?}, \
+            permutations = {permutations:?}\
+        }}",
```

Comment:
```suggestion
            "VerificationKey {{\n"
            "  base_modulus = {base_modulus},\n"
            "  scalar_modulus = {scalar_modulus},\n"
            "  domain = {domain},\n"
            "  cs = {cs},\n"
            "  fixed_commitments = {fixed_commitments:?},\n"
            "  permutations = {permutations:?}\n"
            "}}",
```
Why the double braces, though?

Reply:
I don't think we want newlines in this, because that adds even more dependency on the precise spaces. We should stick as closely as possible to what `#[derive(Debug)]` outputs, which is not pretty-printed.

Reply:
(Clarification: `{:?}` does not pretty-print. `{:#?}` does.)

Reply:
OK, but what is written now *does* include spaces, since this way of doing multiline strings does not strip indentation.

Reply:
Well, default Debug uses spaces AFAIK, just not newlines.

Reply:
It _does_ strip spaces. You can see that in the test case.

Reply:
https://github.com/zcash/halo2/pull/173/files#diff-ce29ac47e4ad4e869b74fd280b52b536e4967606470ae4f64261c50c32e4404fR567

Reply:
@daira https://stackoverflow.com/a/29483453

Reply:
Oh, so it does.